common/containers/Tabs/WalletDecrypt/index.jsx

```diff
@@ -0,0 +1,116 @@
+import React, {Component} from 'react';
+import translate from 'translations';
+import KeystoreDecrypt from './KeystoreDecrypt';
+import PrivateKeyDecrypt from './PrivateKeyDecrypt';
+import MnemonicDecrypt from './MnemonicDecrypt';
+import LedgerNanoSDecrypt from './LedgerNanoSDecrypt';
+import TrezorDecrypt from './TrezorDecrypt';
+import ViewOnlyDecrypt from './ViewOnlyDecrypt';
+
+const decryptionChoices = [
+    {
+        name: 'keystore-file',
+        lid: 'x_Keystore2',
+        component: KeystoreDecrypt
+    },
+    {
+        name: 'private-key',
+        lid: 'x_PrivKey2',
+        component: PrivateKeyDecrypt
+    },
+    {
+        name: 'mnemonic-phrase',
+        lid: 'x_Mnemonic',
+        component: MnemonicDecrypt
+    },
+    {
+        name: 'ledger-nano-s',
+        lid: 'x_Ledger',
+        component: LedgerNanoSDecrypt
+    },
+    {
+        name: 'trezor',
+        lid: 'x_Trezor',
+        component: TrezorDecrypt
+    },
+    {
+        name: 'view-only',
+        lid: 'View with Address Only',
+        component: ViewOnlyDecrypt
+    }
+];
+
+export default class WalletDecrypt extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            decryptionComponent: null
+        };
+    }
+
+    buildDecryptionChoices() {
+        return decryptionChoices.map((decryptionChoice, idx) => {
+            return (
+                <label className="radio" key={decryptionChoice.name}>
+                    <input
+                        aria-flowto={`aria${idx}`}
+                        aria-labelledby={`${decryptionChoice.name}-label`}
+                        type="radio"
+                        name="decryption-choice-radio-group"
+                        value={decryptionChoice.name}
+                        onChange={this.handleDecryptionChoiceChange.bind(this)}
```

Comment:
Ah, nevermind, found the suggested way of handling this scenario. Updated the code and left a comment.